### PR TITLE
Add component version metric

### PR DESCRIPTION
**Pull request by spencerhance**
This metric contains the version of the controller.  It can be overridden with the `INGRESS_VERSION` environment variable.

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -523,3 +539,13 @@ func isServiceFeature(ftr feature) bool {
...
 }
+
+func recordComponentVersion() {
+	var v string
+	v, ok := os.LookupEnv(versionVar)
+	if !ok {
+		klog.V(2).Infof("%q env variable does not exist", versionVar)
+		v = version.Version
+	}
+	componentVersion.WithLabelValues(v).Set(1.0)
```
Comment (swetharepakula) [inline pkg/metrics/metrics.go:550]: nit: make this a constant.

**Review by swetharepakula — COMMENTED** [review]
mostly looks good to me. one small nit

/approve

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (spencerhance) [reply pkg/metrics/metrics.go]: Done

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Review by swetharepakula — COMMENTED** [review]
/lgtm

**Pull request merged**